packages/core/src/utils.ts

```diff
@@ -328,19 +329,42 @@ export const displayProposerTable = (proposerAddresses: string[]) => {
 
 export const displayDeploymentTable = (
   parsedConfig: ParsedChugSplashConfig,
+  artifactPaths: ArtifactPaths,
+  integration: Integration,
   silent: boolean
 ) => {
   if (!silent) {
     const deployments = {}
     Object.entries(parsedConfig.contracts).forEach(
       ([referenceName, contractConfig], i) => {
+        let address = contractConfig.proxy
+
+        // if contract is an unproxied, then we must resolve it's true address
+        if (contractConfig.kind === 'no-proxy') {
+          const { abi, bytecode } = readContractArtifact(
+            artifactPaths[referenceName].contractArtifactPath,
+            integration
+          )
+          const creationCodeWithConstructorArgs =
+            getCreationCodeWithConstructorArgs(
+              bytecode,
+              contractConfig.constructorArgs,
+              referenceName,
+              abi
+            )
+          address = getContractAddress(
+            parsedConfig.options.projectName,
+            creationCodeWithConstructorArgs
+          )
+        }
```

Comment:
I think we can refactor the logic that retrieves `address` for unproxied contracts. It looks like we repeat this pattern of `readContractArtifact` -> `getCreationCodeWithConstructorArgs` -> `getContractAddress` in a few functions: `displayDeploymentTable`, `parseAndValidateChugSplashConfig`, and `foundry/index.ts` (under `getAddress`). If we refactor this into a single function, we can use a ternary operator for the `address`, which IMO is cleaner than how we do it now